src/cpp/shared_core/include/shared_core/Error.hpp

```diff
@@ -735,6 +765,8 @@ Error unknownError(const std::string& in_message, const ErrorLocation& in_locati
  */
 Error unknownError(const std::string& in_message, const Error& in_cause, const ErrorLocation& in_location);
 
+// returns the error message associated with a particular system error code
+std::string systemErrorMessage(int code);
```

Comment:
Can this be entirely defined in the cpp file? Once I pull these changes to the SDK (and release it) it will become very difficult to make signature changes to any functions in public headers (`Error.hpp` is a public header in the SDK).

Reply:
I've made a compromise here:

- `Error.cpp` (in shared_core) gets its own inline version, just so that we avoid changing the public interface for the SDK;
- `System.hpp` (in core) gets a public version, which can be used by other 'core' utilities (and so `LinuxFileMonitor`).

Does that sound reasonable?

Reply:
Ah - I missed that usage in `LinuxFileMonitor`, my mistake. I think it's fine to leave it the other way. I should be able move the declaration to the cpp file with my `pull-shared-core.sh` script. I believe I'm doing that for another function elsewhere already.

Reply:
Done -- thanks!